src/main/java/com/laytonsmith/core/functions/Threading.java

```diff
@@ -586,4 +587,212 @@ public Version since() {
+			return "boolean {string id} Tests if this thread with the given id is alive. A thread is alive if it has been started and has not yet died.";
+		}
+	}
+
+	@api
+	@seealso({x_is_interrupted.class, x_clear_interrupt.class})
+	public static class x_interrupt extends AbstractFunction {
+
+		@Override
+		public Class<? extends CREThrowable>[] thrown() {
+			return null;
+		}
+
+		@Override
+		public boolean isRestricted() {
+			return true;
+		}
+
+		@Override
+		public Boolean runAsync() {
+			return null;
+		}
+
+		@Override
+		public Mixed exec(Target t, Environment environment, Mixed... args) throws ConfigRuntimeException {
+			Thread th;
+			if(args.length == 1) {
+				th = THREAD_ID_MAP.get(args[0].val());
+				if(th != null) {
+					th.interrupt();
+				}
+			} else {
+				Thread.currentThread().interrupt();
+			}
+			return CVoid.VOID;
+		}
+
+		@Override
+		public Version since() {
+			return MSVersion.V3_3_4;
+		}
+
+		@Override
+		public String getName() {
+			return "x_interrupt";
+		}
+
+		@Override
+		public Integer[] numArgs() {
+			return new Integer[]{0, 1};
+		}
+
+		@Override
+		public String docs() {
+			return "void {[string id]} Interrupts the thread with the given id, or the current thread if no id is given."
+					+ " When a thread is interrupted, its interrupt status is set to true."
+					+ " This status can be checked using the " + new x_is_interrupted().getName() + " and " + new x_clear_interrupt().getName() + " function."
+					+ " Note that some blocking functions will throw an InterruptedException when the thread on which they are executed is interrupted.";
+		}
+	}
+
+	@api
+	@seealso({x_interrupt.class, x_clear_interrupt.class})
+	public static class x_is_interrupted extends AbstractFunction {
+
+		@Override
+		public Class<? extends CREThrowable>[] thrown() {
+			return null;
+		}
+
+		@Override
+		public boolean isRestricted() {
+			return true;
+		}
+
+		@Override
+		public Boolean runAsync() {
+			return null;
+		}
+
+		@Override
+		public Mixed exec(Target t, Environment environment, Mixed... args) throws ConfigRuntimeException {
+			if(args.length == 1) {
+				Thread th = THREAD_ID_MAP.get(args[0].val());
+				if(th != null) {
+					return CBoolean.get(th.isInterrupted());
+				} else {
+					return CBoolean.FALSE;
+				}
+			} else {
+				return CBoolean.get(Thread.currentThread().isInterrupted());
+			}
+		}
+
+		@Override
+		public Version since() {
+			return MSVersion.V3_3_4;
+		}
+
+		@Override
+		public String getName() {
+			return "x_is_interrupted";
+		}
+
+		@Override
+		public Integer[] numArgs() {
+			return new Integer[]{0, 1};
+		}
+
+		@Override
+		public String docs() {
+			return "boolean {[string id]} Tests whether this thread with the given id has been interrupted."
+					+ " If id not passed, it is be current thread."
+					+ " The \"interrupt status\" of the thread is unaffected by this method."
+					+ " In case if the thread does not exist or is not alive, return false";
```

Comment:
"Tests whether the thread with the given id (or the current thread if none is provided) is interrupted via x_interrupt. If the thread doesn’t exist, or is not alive, false is returned."

Let's not mention the thing about it affecting the interrupt state. That's a weird java quirk that's irrelevant in MethodScript.

Reply:
So delete all mentions of interrupt status?

Reply:
Just use the string I provided (but pull the function name dynamically).